### DISCUSSION TITLE
Remove silence when alert is cleared

### DISCUSSION THREAD

**Discussion by tom0010** [grafana/grafana#39615] [Alerting]
The alerts I have setup are being sent to a webhook, which in turn is going to slack via a bot.
I've built a button inside slack to silence the alerts using the grafana/alertmanager API, which works fine. 
However, when an alert is cleared it seems the silence is still present. 
Is there a way we could add this functionality to clear the silence if the alert is cleared? 

I have currently built something using the API with python/flask as a workaround but, it's not really ideal because it doesn't send the "resolved" alerts every time, especially if the alert is silenced.
```
@app.route("/alertmanager/alerts/clear", methods=["POST"])
def alertmanager_clear():
    if request.args.get("clear_alerts") == "true":
        return alertmanager.clear_alerts()
```

```
URL = "http://grafana:3000/api/alertmanager/grafana/api/v2"

def clear_alerts():
    headers = {"Authorization": "Bearer token"}
    url = f"{URL}/alerts"
    alerts = requests.get(url, headers=headers).json()
    alert_ids = [alert["status"]["silencedBy"][0] for alert in alerts if len(alert["status"]["silencedBy"]) > 0]

    url = f"{URL}/silences"
    silences = requests.get(url, headers=headers).json()
    silence_ids = [silence["id"] for silence in silences if silence["status"]["state"] != "expired"]

    delete_ids = [delete for delete in silence_ids if delete not in alert_ids]
    url = f"{URL}/silence"
    for delete_id in delete_ids:
        requests.delete(f"{url}/delete_id", headers=headers)
    return "OK", 200
```

To "test" and get this working, I'm simply just using a while loop to do a POST response on that new endpoint:
```
while true; do curl -X POST http://my_host:my_port/alertmanager/alerts/clear?clear_alerts=true; sleep 1; done
```
A "workaround" but not the best solution :) 
Thanks

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
